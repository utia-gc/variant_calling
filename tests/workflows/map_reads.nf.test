nextflow_workflow {

    name "Test Workflow MAP_READS"
    script "workflows/map_reads.nf"
    workflow "MAP_READS"

    test("MAP_READS maps reads to a reference genome and outputs sorted BAMs -- bwa-mem2 + SE reads.") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()
        def readsLane2 = new ReadsSELane2()
        def refs       = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [
                            sampleName:   "${readsLane1.getSampleName()}",
                            sampleNumber: "${readsLane1.getSampleNumber()}",
                            lane:         "${readsLane1.getLane()}",
                            readType:     "${readsLane1.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane1.getRGFields().get('ID')}",
                                SM: "${readsLane1.getRGFields().get('SM')}",
                                LB: "${readsLane1.getRGFields().get('LB')}",
                                PL: "${readsLane1.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane1.getR1()}"),
                        file("${readsLane1.getR2()}")
                    ],
                    [
                        [
                            sampleName:   "${readsLane2.getSampleName()}",
                            sampleNumber: "${readsLane2.getSampleNumber()}",
                            lane:         "${readsLane2.getLane()}",
                            readType:     "${readsLane2.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane2.getRGFields().get('ID')}",
                                SM: "${readsLane2.getRGFields().get('SM')}",
                                LB: "${readsLane2.getRGFields().get('LB')}",
                                PL: "${readsLane2.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane2.getR1()}"),
                        file("${readsLane2.getR2()}")
                    ]
                )
                input[1] = file("${refs.getGenome()}")
                input[2] = file("${refs.getGTF()}")
                input[3] = "bwamem2"
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test alignments for each individual coordinate sorted sample-lane
            with(workflow.out.alignmentsIndividualSortedByCoord) {
                assert size() == 2
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}\.bam\.bai/
                }
                with(get(1)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane2.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane2.getStemName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and coordinate sorted sample
            with(workflow.out.alignmentsMergedSortedByCoord) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getSampleName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and name sorted sample
            with(workflow.out.alignmentsMergedSortedByName) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 2
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                }
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

    test("MAP_READS maps reads to a reference genome and outputs sorted BAMs -- bwa-mem2 + PE reads.") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()
        def readsLane2 = new ReadsPELane2()
        def refs       = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [
                            sampleName:   "${readsLane1.getSampleName()}",
                            sampleNumber: "${readsLane1.getSampleNumber()}",
                            lane:         "${readsLane1.getLane()}",
                            readType:     "${readsLane1.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane1.getRGFields().get('ID')}",
                                SM: "${readsLane1.getRGFields().get('SM')}",
                                LB: "${readsLane1.getRGFields().get('LB')}",
                                PL: "${readsLane1.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane1.getR1()}"),
                        file("${readsLane1.getR2()}")
                    ],
                    [
                        [
                            sampleName:   "${readsLane2.getSampleName()}",
                            sampleNumber: "${readsLane2.getSampleNumber()}",
                            lane:         "${readsLane2.getLane()}",
                            readType:     "${readsLane2.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane2.getRGFields().get('ID')}",
                                SM: "${readsLane2.getRGFields().get('SM')}",
                                LB: "${readsLane2.getRGFields().get('LB')}",
                                PL: "${readsLane2.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane2.getR1()}"),
                        file("${readsLane2.getR2()}")
                    ]
                )
                input[1] = file("${refs.getGenome()}")
                input[2] = file("${refs.getGTF()}")
                input[3] = "bwamem2"
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test alignments for each individual coordinate sorted sample-lane
            with(workflow.out.alignmentsIndividualSortedByCoord) {
                assert size() == 2
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}\.bam\.bai/
                }
                with(get(1)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane2.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane2.getStemName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and coordinate sorted sample
            with(workflow.out.alignmentsMergedSortedByCoord) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getSampleName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and name sorted sample
            with(workflow.out.alignmentsMergedSortedByName) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 2
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                }
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

    test("MAP_READS maps reads to a reference genome and outputs sorted BAMs -- STAR + SE reads.") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()
        def readsLane2 = new ReadsSELane2()
        def refs       = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [
                            sampleName:   "${readsLane1.getSampleName()}",
                            sampleNumber: "${readsLane1.getSampleNumber()}",
                            lane:         "${readsLane1.getLane()}",
                            readType:     "${readsLane1.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane1.getRGFields().get('ID')}",
                                SM: "${readsLane1.getRGFields().get('SM')}",
                                LB: "${readsLane1.getRGFields().get('LB')}",
                                PL: "${readsLane1.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane1.getR1()}"),
                        file("${readsLane1.getR2()}")
                    ],
                    [
                        [
                            sampleName:   "${readsLane2.getSampleName()}",
                            sampleNumber: "${readsLane2.getSampleNumber()}",
                            lane:         "${readsLane2.getLane()}",
                            readType:     "${readsLane2.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane2.getRGFields().get('ID')}",
                                SM: "${readsLane2.getRGFields().get('SM')}",
                                LB: "${readsLane2.getRGFields().get('LB')}",
                                PL: "${readsLane2.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane2.getR1()}"),
                        file("${readsLane2.getR2()}")
                    ]
                )
                input[1] = file("${refs.getGenome()}")
                input[2] = file("${refs.getGTF()}")
                input[3] = "star"
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test alignments for each individual coordinate sorted sample-lane
            with(workflow.out.alignmentsIndividualSortedByCoord) {
                assert size() == 2
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}\.bam\.bai/
                }
                with(get(1)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane2.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane2.getStemName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and coordinate sorted sample
            with(workflow.out.alignmentsMergedSortedByCoord) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getSampleName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and name sorted sample
            with(workflow.out.alignmentsMergedSortedByName) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 2
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                }
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

    test("MAP_READS maps reads to a reference genome and outputs sorted BAMs -- STAR + PE reads.") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()
        def readsLane2 = new ReadsPELane2()
        def refs       = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [
                            sampleName:   "${readsLane1.getSampleName()}",
                            sampleNumber: "${readsLane1.getSampleNumber()}",
                            lane:         "${readsLane1.getLane()}",
                            readType:     "${readsLane1.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane1.getRGFields().get('ID')}",
                                SM: "${readsLane1.getRGFields().get('SM')}",
                                LB: "${readsLane1.getRGFields().get('LB')}",
                                PL: "${readsLane1.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane1.getR1()}"),
                        file("${readsLane1.getR2()}")
                    ],
                    [
                        [
                            sampleName:   "${readsLane2.getSampleName()}",
                            sampleNumber: "${readsLane2.getSampleNumber()}",
                            lane:         "${readsLane2.getLane()}",
                            readType:     "${readsLane2.getReadType()}",
                            rgFields:     [
                                ID: "${readsLane2.getRGFields().get('ID')}",
                                SM: "${readsLane2.getRGFields().get('SM')}",
                                LB: "${readsLane2.getRGFields().get('LB')}",
                                PL: "${readsLane2.getRGFields().get('PL')}",
                            ]
                        ],
                        file("${readsLane2.getR1()}"),
                        file("${readsLane2.getR2()}")
                    ]
                )
                input[1] = file("${refs.getGenome()}")
                input[2] = file("${refs.getGTF()}")
                input[3] = "star"
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test alignments for each individual coordinate sorted sample-lane
            with(workflow.out.alignmentsIndividualSortedByCoord) {
                assert size() == 2
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}\.bam\.bai/
                }
                with(get(1)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane2.getStemName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane2.getStemName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and coordinate sorted sample
            with(workflow.out.alignmentsMergedSortedByCoord) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 3
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                    // test indexed alignments file
                    assert get(2) ==~ /^.*\/${readsLane1.getSampleName()}\.bam\.bai/
                }
            }
            // test alignments for each merged and name sorted sample
            with(workflow.out.alignmentsMergedSortedByName) {
                assert size() == 1
                with(get(0)) {
                    assert size() == 2
                    // test alignments file
                    assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.bam/
                }
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

}
