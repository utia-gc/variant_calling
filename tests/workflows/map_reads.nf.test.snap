{
    "MAP_READS maps reads to a reference genome and outputs sorted BAMs -- STAR + SE reads.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "001",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.001",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L001.bam:md5,84a6259469927439f1c1b120e77fa5d8",
                        "SRR1066657_L001.bam.bai:md5,fa09355ba8126b7666351838eff70751"
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "002",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.002",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L002.bam:md5,7abeac9fe62740928701db106d5a8f80",
                        "SRR1066657_L002.bam.bai:md5,6151f3a8699b203e24e92daf69af0d7e"
                    ]
                ],
                "1": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,fc39fecfefdf25bedd72c1719fbd5430",
                        "SRR1066657.bam.bai:md5,4e376217dd435f6f02a1e981f76bdcde"
                    ]
                ],
                "2": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,2cbb62cebe1ab7443d4c6bdf311f2f54"
                    ]
                ],
                "alignmentsIndividualSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "001",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.001",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L001.bam:md5,84a6259469927439f1c1b120e77fa5d8",
                        "SRR1066657_L001.bam.bai:md5,fa09355ba8126b7666351838eff70751"
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "002",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.002",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L002.bam:md5,7abeac9fe62740928701db106d5a8f80",
                        "SRR1066657_L002.bam.bai:md5,6151f3a8699b203e24e92daf69af0d7e"
                    ]
                ],
                "alignmentsMergedSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,fc39fecfefdf25bedd72c1719fbd5430",
                        "SRR1066657.bam.bai:md5,4e376217dd435f6f02a1e981f76bdcde"
                    ]
                ],
                "alignmentsMergedSortedByName": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,2cbb62cebe1ab7443d4c6bdf311f2f54"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-08T12:59:22.10289253"
    },
    "MAP_READS maps reads to a reference genome and outputs sorted BAMs -- bwa-mem2 + SE reads.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "001",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.001",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L001.bam:md5,4127b2dd883040a440d62b90ca52cf65",
                        "SRR1066657_L001.bam.bai:md5,310e773f12ef2ee2041417625519058c"
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "002",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.002",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L002.bam:md5,cf4a89405d61a82c011e17533ac22116",
                        "SRR1066657_L002.bam.bai:md5,feed2c18a1adcc626a92d3f25d6c8fdb"
                    ]
                ],
                "1": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,e9adf40bdbe36f51bcb843241080858e",
                        "SRR1066657.bam.bai:md5,3077e794d9d77128b8a5c88fd5e0887a"
                    ]
                ],
                "2": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,e6b0b440a1ecc998d762e22fecafaebd"
                    ]
                ],
                "alignmentsIndividualSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "001",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.001",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L001.bam:md5,4127b2dd883040a440d62b90ca52cf65",
                        "SRR1066657_L001.bam.bai:md5,310e773f12ef2ee2041417625519058c"
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "002",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.002",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L002.bam:md5,cf4a89405d61a82c011e17533ac22116",
                        "SRR1066657_L002.bam.bai:md5,feed2c18a1adcc626a92d3f25d6c8fdb"
                    ]
                ],
                "alignmentsMergedSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,e9adf40bdbe36f51bcb843241080858e",
                        "SRR1066657.bam.bai:md5,3077e794d9d77128b8a5c88fd5e0887a"
                    ]
                ],
                "alignmentsMergedSortedByName": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "readType": "single"
                        },
                        "SRR1066657.bam:md5,e6b0b440a1ecc998d762e22fecafaebd"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-08T12:57:45.055725698"
    },
    "MAP_READS maps reads to a reference genome and outputs sorted BAMs -- bwa-mem2 + PE reads.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.001",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L001.bam:md5,34739d31119b05ae752e830d60a1a211",
                        "SRR6924569_L001.bam.bai:md5,1e9ba5912083c49a1cd61a4556b3b7e4"
                    ],
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "002",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.002",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L002.bam:md5,91ffdeef04cc4e98d935e0ddf58c9575",
                        "SRR6924569_L002.bam.bai:md5,70e5ded34544598a2f21ca31ad56179b"
                    ]
                ],
                "1": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,96909be9160cad915e415097f4eccd49",
                        "SRR6924569.bam.bai:md5,e4dd145a9dddbb93dd4747696c68a6e0"
                    ]
                ],
                "2": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,d1fda3c5b77b5ff86707b4c20f87ede9"
                    ]
                ],
                "alignmentsIndividualSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.001",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L001.bam:md5,34739d31119b05ae752e830d60a1a211",
                        "SRR6924569_L001.bam.bai:md5,1e9ba5912083c49a1cd61a4556b3b7e4"
                    ],
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "002",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.002",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L002.bam:md5,91ffdeef04cc4e98d935e0ddf58c9575",
                        "SRR6924569_L002.bam.bai:md5,70e5ded34544598a2f21ca31ad56179b"
                    ]
                ],
                "alignmentsMergedSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,96909be9160cad915e415097f4eccd49",
                        "SRR6924569.bam.bai:md5,e4dd145a9dddbb93dd4747696c68a6e0"
                    ]
                ],
                "alignmentsMergedSortedByName": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,d1fda3c5b77b5ff86707b4c20f87ede9"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-08T12:58:14.233499615"
    },
    "MAP_READS maps reads to a reference genome and outputs sorted BAMs -- STAR + PE reads.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.001",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L001.bam:md5,6c15cc278e607bb09f3755fe928c659e",
                        "SRR6924569_L001.bam.bai:md5,3ae744447ddbb4ed67235d29dcaa6b64"
                    ],
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "002",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.002",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L002.bam:md5,3f8d51e4ff0e5ac57c88d5832503f2ef",
                        "SRR6924569_L002.bam.bai:md5,e9d7771744655c8ecc5b1c64b454c765"
                    ]
                ],
                "1": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,513fd7c77e7e7052174012cb3d81dd19",
                        "SRR6924569.bam.bai:md5,7170e2828d1d5c940e6a4b1cd38516f3"
                    ]
                ],
                "2": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,2ec2c9a556f802f9310a3d97f7504e27"
                    ]
                ],
                "alignmentsIndividualSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.001",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L001.bam:md5,6c15cc278e607bb09f3755fe928c659e",
                        "SRR6924569_L001.bam.bai:md5,3ae744447ddbb4ed67235d29dcaa6b64"
                    ],
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "002",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.002",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L002.bam:md5,3f8d51e4ff0e5ac57c88d5832503f2ef",
                        "SRR6924569_L002.bam.bai:md5,e9d7771744655c8ecc5b1c64b454c765"
                    ]
                ],
                "alignmentsMergedSortedByCoord": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,513fd7c77e7e7052174012cb3d81dd19",
                        "SRR6924569.bam.bai:md5,7170e2828d1d5c940e6a4b1cd38516f3"
                    ]
                ],
                "alignmentsMergedSortedByName": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "readType": "paired"
                        },
                        "SRR6924569.bam:md5,2ec2c9a556f802f9310a3d97f7504e27"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-08T13:03:16.211113597"
    }
}